src/Marquee.tsx

```diff
@@ -188,24 +182,20 @@ export class Marquee extends Component<Props, State> {
     }
 
     private measureRuntimeVariablesAgainIfWindowIsResized(): void {
-        this.resizeCallback = (): void => {
-            const pageWidthHasChanged: boolean = this.pageWidth !== Marquee.getPageWidth();
-
-            if (pageWidthHasChanged) {
-                this.stopAnimation();
-                this.setupRuntimeVariables();
-                this.startAnimationAndForceReflow();
-            }
-        };
+        this.resizeCallback = new window.ResizeObserver(() => {
+            this.stopAnimation();
+            this.setupRuntimeVariables();
+            this.startAnimationAndForceReflow();
+        });
 
-        window.addEventListener('resize', this.resizeCallback);
+        this.resizeCallback.observe(this.marqueeElement);
     }
 
     // This is to force a reflow which is necessary in order to transition styles when adding a class name.
-    private startAnimationAndForceReflow() {
+    private startAnimationAndForceReflow(): void {
         setTimeout(() => {
```

Comment:
Do you still need settimeout here?

Reply:
nope, works even without stopping and restarting the animation. will remove that functionallity.